crates/core/src/client/mod.rs

```diff
@@ -483,43 +483,9 @@ impl KakarotProvider for KakarotClient<JsonRpcClient<HttpTransport>> {
 
                     // Cannot use `map` because of the `await` call.
                     for event in events {
-                        let contract_address = self.safe_get_evm_address(&event.from_address, &starknet_block_id).await;
-
-                        // event "keys" in cairo are event "topics" in solidity
-                        // they're returned as list where consecutive values are
-                        // low, high, low, high, etc. of the Uint256 Cairo representation
-                        // of the bytes32 topics. This recomputes the original topic
-                        let topics = (0..event.keys.len())
-                            .step_by(2)
-                            .map(|i| {
-                                let next_key = *event.keys.get(i + 1).unwrap_or(&FieldElement::ZERO);
-
-                                // Can unwrap here as we know 2^128 is a valid FieldElement
-                                let two_pow_16: FieldElement =
-                                    FieldElement::from_hex_be("0x100000000000000000000000000000000").unwrap();
-
-                                // TODO: May wrap around prime field - Investigate edge cases
-                                let felt_shifted_next_key = next_key * two_pow_16;
-                                event.keys[i] + felt_shifted_next_key
-                            })
-                            .map(|topic| H256::from(&topic.to_bytes_be()))
-                            .collect::<Vec<_>>();
-
-                        let data = vec_felt_to_bytes(event.data);
-
-                        let log = Log {
-                            // TODO: fetch correct address from Kakarot.
-                            // Contract Address is the account contract's address (EOA or KakarotAA)
-                            address: H160::from_slice(&contract_address.0),
-                            topics,
-                            data: RpcBytes::from(data.0),
-                            block_hash: None,
-                            block_number: None,
-                            transaction_hash: None,
-                            transaction_index: None,
-                            log_index: None,
-                            removed: false,
-                        };
+                        let event = StarknetEvent::new(event);
+                        let log =
+                            event.to_eth_log(self, block_hash, block_number, transaction_hash, None, None).await?;
```

Comment:
Yes, it makes sense, I have refactored.

- So, I also noticed that the mock response for the RPC call doesn't need to be changed! Rather than throwing for this event, it should get filtered out { initially I thought that was a mistake }. I have reverted that file back to the old state.
- As we discussed that in this case any error from `to_eth_log` represents the fact that the log needs to be filtered out, so I have refactored the code accordingly.